### PR TITLE
refactor: optimize user list query

### PR DESCRIPTION
**Pull request by ishanarya0**

### CODE REVIEW

File: internal/store/postgres/user_repository.go [hunk 1]
```diff
@@ -277,14 +278,44 @@ func (r UserRepository) List(ctx context.Context, flt user.Filter) ([]user.User,
...
+			fmt.Sprintf("%s.email", TABLE_USERS),
+			fmt.Sprintf("%s.key", TABLE_METADATA),
+			fmt.Sprintf("%s.value", TABLE_METADATA),
+			fmt.Sprintf("%s.created_at", TABLE_USERS),
+			fmt.Sprintf("%s.updated_at", TABLE_USERS),
+		).Limit(uint(flt.Limit)).Offset(uint(offset)).ToSQL()
+		if err != nil {
+			return []user.User{}, fmt.Errorf("%w: %s", queryErr, err)
+		}
 	}
```
Comment (mabdh) [inline internal/store/postgres/user_repository.go]: Can we do this instead to shorten it?
```suggestion
  	var selectedUsers *goqu.SelectDataset

	if flt.Keyword != "" {
		selectedUsers = dialect.From(TABLE_USERS).Where(goqu.Or(
			goqu.C("name").ILike(fmt.Sprintf("%%%s%%", flt.Keyword)),
			goqu.C("email").ILike(fmt.Sprintf("%%%s%%", flt.Keyword))),
		).As(FILTERED_USERS)
	} else {
		selectedUsers = dialect.From(TABLE_USERS).As(FILTERED_USERS)
	}

	query, params, err = selectedUsers.LeftOuterJoin(
		goqu.T(TABLE_METADATA),
		goqu.On(goqu.Ex{fmt.Sprintf("%s.id", FILTERED_USERS): goqu.I(fmt.Sprintf("%s.user_id", TABLE_METADATA))})).Select(
		fmt.Sprintf("%s.id", FILTERED_USERS),
		fmt.Sprintf("%s.name", FILTERED_USERS),
		fmt.Sprintf("%s.email", FILTERED_USERS),
		fmt.Sprintf("%s.key", TABLE_METADATA),
		fmt.Sprintf("%s.value", TABLE_METADATA),
		fmt.Sprintf("%s.created_at", FILTERED_USERS),
		fmt.Sprintf("%s.updated_at", FILTERED_USERS),
	).Limit(uint(flt.Limit)).Offset(uint(offset)).ToSQL()
	if err != nil {
		return []user.User{}, fmt.Errorf("%w: %s", queryErr, err)
	}
```

**Comment by mabdh** [comment]
@ishanarya0 can you share the results of `EXPLAIN ANALYZE` of both approaches? Why the changed code is more optimize?

File: internal/store/postgres/user_repository.go (hunk 1 above)
Comment (ishanarya0) [reply internal/store/postgres/user_repository.go]: Yes we can

**Comment by ishanarya0** [comment]
> @ishanarya0 can you share the results of `EXPLAIN ANALYZE` of both approaches? Why the changed code is more optimize?

sure

**Review by mabdh — APPROVED** [review]

**Pull request merged**